### PR TITLE
Changed free drag to ordered draggable list

### LINKED ISSUE #234: Change timeline pages from free drag to draggable ordered list

### PR DESCRIPTION
**Pull request by lauriLukkarinen**
- also fixed couple small warnings and errors

Resolves #234

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "material-ui-dropzone": "^3.0.0",
     "moment": "^2.24.0",
     "react": "^16.12.0",
+    "react-beautiful-dnd": "^13.0.0",
```
Comment (jarinykanen) [inline package.json:36]: Remove old library

File: src/components/content-editor/timeline-editor.tsx [hunk 2]
```diff
@@ -36,44 +50,116 @@ const TimelineEditor: React.FC<Props> = (props: Props) => {
...
 /**
  * Renders page as draggable item
  * 
  * @param page page
  * @param props component props
  */
-const renderDraggableItem = (page: ExhibitionPage, props: Props) => {
-  const { onClick, selectedPage, classes } = props;
-  const selected = selectedPage?.id === page.id;
+const renderDraggablePage = (index: number, page: ExhibitionPage, props: Props) => {
```
Comment (jarinykanen) [inline src/components/content-editor/timeline-editor.tsx:113]: Fix docs

File: src/components/screens/content-editor-screen.tsx [hunk 3]
```diff
@@ -722,6 +749,51 @@ class ContentEditorScreen extends React.Component<Props, State> {
     });
   }
 
+  /**
+   * Event handler for page drag end
+   * 
+   * @param deviceId device id
+   */
+  private onPageDragEnd = (deviceId: string) => (result: DropResult) => {
```
Comment (jarinykanen) [inline src/components/screens/content-editor-screen.tsx:757]: Add param

File: src/components/screens/content-editor-screen.tsx [hunk 4]
```diff
@@ -722,6 +749,51 @@ class ContentEditorScreen extends React.Component<Props, State> {
...
+  }
+
+  /**
+   * Reorders device pages
+   * 
+   * @param pageOrder page order
+   * @param prevIndex previous index
+   * @param newIndex new index
+   */
+  private reorderPages = (pageOrder: string[], prevIndex: number, newIndex: number): string[] => {
```
Comment (jarinykanen) [inline src/components/screens/content-editor-screen.tsx:789]: Add return docs

**Review by jarinykanen — CHANGES_REQUESTED** [review]
Fix comments

File: package.json (hunk 1 above)
Comment (lauriLukkarinen) [reply package.json:36]: This is the new one, but good point, I'll remove the old nevertheless.

**Pull request merged**